File: CivitasR/lib/juego_texto.rb
#encoding:utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "civitas_juego"
require_relative "vista_textual"
require_relative "controlador"

module Civitas
  class Juego_texto
    
    def self.main

      jugadores = Array.new

      print "Indica el número de jugadores: "
      num_jugadores = gets.chomp
      
      i = 0
      
      print "Indica el nombre de los jugadores:\n"
      
      loop do
        
        print "Nombre jugador " + i.to_s + ":"
        jugadores.push(gets.chomp)
        
        i = i + 1
        
        break if (i >= num_jugadores.to_i)
      end
      juego = Civitas_Juego.new(jugadores)
      vista = Vista_textual.new()
      dado = Dado.instance
      dado.set_debug(true)
      controlador = Controlador.new(juego, vista)
      controlador.juega()
      
    end
  end
  
  Juego_texto.main

end

File: CivitasJ/src/civitas/JuegoTexto.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;
import java.util.Scanner;
import juegoTexto.*;

/**
 *
 * @author ramonjtorres
 */
public class JuegoTexto {
    
    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
    
        Scanner sc = new Scanner(System.in);
        int num_jugadores;
        
        do{
            System.out.println("Indique el número de jugadores (de 2 a 4 jugadores): ");
            num_jugadores = sc.nextInt();
        
        }while(num_jugadores < 2 || num_jugadores > 4);
        
        ArrayList<String> jugadores = new ArrayList();
        int i = 0;
        
        System.out.println("Indique el nombre de los jugadores:");
        
        do{
        
            System.out.println("Nombre jugador " + i + ": ");
            jugadores.add(sc.next());
            i++;
        }while( i < num_jugadores);
        
        System.out.println("================================= COMENZAMOS ==================================");

        CivitasJuego juego = new CivitasJuego(jugadores);
        VistaTextual vista = new VistaTextual();
        Dado dado = Dado.getInstance();
        dado.setDebug(true);
        Controlador controlador = new Controlador(juego, vista);
        controlador.juega();
    }
}
